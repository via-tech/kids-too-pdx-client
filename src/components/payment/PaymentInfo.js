import React from 'react';
import PropTypes from 'prop-types';
import styles from './Registration.css';

function PaymentInfo({ handleChange, subFee }) {
  return (
    <>
      <fieldset>
        <legend>Payment Info</legend>

        <label>Card Number<span className={styles.required}>*</span>
          <input type="text" name="cardNumber" pattern="[0-9]*" maxLength="19" onChange={handleChange} placeholder="Card Number" required />
        </label>

        <label>Security Code<span className={styles.required}>*</span>
          <input type="text" name="securityCode" maxLength="3" onChange={handleChange} required />
        </label>

        <label>Name<span className={styles.required}>*</span>
          <input type="text" name="cardName" placeholder="Card Name" onChange={handleChange} required />
        </label>

        <label>Expiration Date<span className={styles.required}>*</span>
          <input type="text" name="expMonth" pattern="[0-9]*" placeholder="mm" maxLength="2" onChange={handleChange} required />

          <input type="text" name="expYear" pattern="[0-9]*" placeholder="yyyy" minLength="4" maxLength="4" onChange={handleChange} required />
        </label>

        <label>Method<span className={styles.required}>*</span>
          <input type="radio" name="method" value="visa" onChange={handleChange} required />Visa

          <input type="radio" name="method" value="master" onChange={handleChange} required />Master
        </label>

        <label>Billing Address <span className={styles.required}>*</span>
          <label>Street:
            <input type="text" name="billStreet" onChange={handleChange} />
          </label>

          <label>City:
            <input type="text" name="billCity" onChange={handleChange} />
          </label>

          <label>State
            <input type="text" name="state" onChange={handleChange} required />
          </label>

          <label>Zip Code:
            <input type="text" name="billZipcode" pattern="[0-9]*" onChange={handleChange} />
          </label>
        </label>
      </fieldset>

      <h2>Total: ${subFee}</h2>
    </>
  );
}

PaymentInfo.propTypes = {
  handleChange: PropTypes.func,
  subFee: PropTypes.string
};

export default PaymentInfo;
