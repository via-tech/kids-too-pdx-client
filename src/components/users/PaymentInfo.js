import React from 'react';
import PropTypes from 'prop-types';

function PaymentInfo({ handleChange }) {
  return (
    <>
      <fieldset>
        <legend>Payment Info</legend>

        <label>Card Number
          <input type="text" name="cardNumber" pattern="[0-9]*" maxLength="19" onChange={handleChange} placeholder="Card Number" required />
        </label>

        <label>Security Code
          <input type="text" name="securityCode" maxLength="3" onChange={handleChange} required />
        </label>

        <label>Name
          <input type="text" name="cardName" placeholder="Card Name" onChange={handleChange} required />
        </label>

        <label>Expiration Date
          <input type="text" name="expMonth" pattern="[0-9]*" placeholder="mm" maxLength="2" onChange={handleChange} required />

          <input type="text" name="expYear" pattern="[0-9]*" placeholder="yyyy" minLength="4" maxLength="4" onChange={handleChange} required />
        </label>

        <label>Method
          <input type="radio" name="method" value="visa" onChange={handleChange} required />Visa

          <input type="radio" name="method" value="master" onChange={handleChange} required />Master
        </label>

        <label>Billing Address (if different from business address)
          <label>Street:
            <input type="text" name="billStreet" onChange={handleChange} />
          </label>

          <label>City:
            <input type="text" name="billCity" onChange={handleChange} />
          </label>

          <label>Zip Code:
            <input type="text" name="billZipcode" pattern="[0-9]*" onChange={handleChange} required />
          </label>
        </label>
      </fieldset>

      <h2>Total: $1.00</h2>
    </>
  );
}

PaymentInfo.propTypes = {
  handleChange: PropTypes.func
};

export default PaymentInfo;
