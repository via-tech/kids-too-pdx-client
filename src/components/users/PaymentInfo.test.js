import React from 'react';
import renderer from 'react-test-renderer';
import PaymentInfo from './PaymentInfo';

describe('Registration Form', () => {
  it('snapshots PaymentInfo form', () => {
    const props = {
      handleChange: jest.fn()
    };

    const tree = renderer.create(
      <PaymentInfo {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
