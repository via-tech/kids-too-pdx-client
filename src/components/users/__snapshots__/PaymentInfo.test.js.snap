// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Form snapshots PaymentInfo form 1`] = `
Array [
  <fieldset>
    <legend>
      Payment Info
    </legend>
    <label>
      Card Number
      <input
        maxLength="19"
        name="cardNumber"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="Card Number"
        required={true}
        type="text"
      />
    </label>
    <label>
      Security Code
      <input
        maxLength="3"
        name="securityCode"
        onChange={[MockFunction]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Name
      <input
        name="cardName"
        onChange={[MockFunction]}
        placeholder="Card Name"
        required={true}
        type="text"
      />
    </label>
    <label>
      Expiration Date
      <input
        maxLength="2"
        name="expMonth"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="mm"
        required={true}
        type="text"
      />
      <input
        maxLength="4"
        minLength="4"
        name="expYear"
        onChange={[MockFunction]}
        pattern="[0-9]*"
        placeholder="yyyy"
        required={true}
        type="text"
      />
    </label>
    <label>
      Method
      <input
        name="method"
        onChange={[MockFunction]}
        required={true}
        type="radio"
        value="visa"
      />
      Visa
      <input
        name="method"
        onChange={[MockFunction]}
        required={true}
        type="radio"
        value="master"
      />
      Master
    </label>
    <label>
      Billing Address (if different from business address)
      <label>
        Street:
        <input
          name="billStreet"
          onChange={[MockFunction]}
          type="text"
        />
      </label>
      <label>
        City:
        <input
          name="billCity"
          onChange={[MockFunction]}
          type="text"
        />
      </label>
      <label>
        Zip Code:
        <input
          name="billZipcode"
          onChange={[MockFunction]}
          pattern="[0-9]*"
          required={true}
          type="text"
        />
      </label>
    </label>
  </fieldset>,
  <h2>
    Total: $1.00
  </h2>,
]
`;
